apps/blc/blc.js

```diff
@@ -2,135 +2,181 @@
-    
+    const SETTINGSFILE = "BinaryClk.settings.json";
+
+    // variables defined from settings
+    let HourCol;
+    let MinCol;
+    let DayCol;
+    let MonCol;
+    let RingOn;
+
+    // color arrays
+    // !!! don't change order unless change oder in BinaryClk.settings.js !!!
+    // !!! order must correspond to each other between arrays !!!
+    let LED_Colors =              ["#FFF", "#F00", "#0F0",  "#00F", "#FF0",    "#F0F",        "#0FF", "#000"];
+    let LED_ColorNames = ["white", "red",     "green", "blue",   "yellow", "magenta", "cyan",   "black"];   
 
-////////////////////////////////////////
-// compute bit-pattern from time/date and draw leds
-////////////////////////////////////////  
-    var line_cnt = 0;
-    var cnt = 0;
-    var bit_cnt = 0;
+    // load settings
+    let loadSettings = function()
+    {
+        function def (value, def) {return value !== undefined ? value : def;}
+
+        var settings = require('Storage').readJSON(SETTINGSFILE, true) || {};
+        // get name from setting, find index of name and assign corresponding color code by index
+        HourCol = LED_Colors[LED_ColorNames.indexOf(def(settings.HourCol, "red"))]; 
+        MinCol =  LED_Colors[LED_ColorNames.indexOf(def(settings.MinCol, "green"))]; 
+        DayCol = LED_Colors[LED_ColorNames.indexOf(def(settings.DayCol, "yellow"))];
+        MonCol = LED_Colors[LED_ColorNames.indexOf(def(settings.MonCol, "yellow"))];
+        RingOn = def(settings.RingOn, true);
     
-    while (line_cnt < nr_lines)
-      {
-        
-        ////////////////////////////////////////
-        // compute bit-pattern
-        bit_cnt = msbits[line_cnt];
+        delete settings;    // settings in local var -> no more required
+    }
 
-        while (bit_cnt >= 0)
+    let drawTimeout;
+
+    // actually draw the watch face
+    let draw = function()
+    {
+        // Bangle.js2 -> 176x176
+        var x_rgt = g.getWidth();
+        var y_bot = g.getHeight();
+        //var x_cntr = x_rgt / 2;
+        var y_cntr = y_bot / 18*7;      
+        g.reset().clearRect(Bangle.appRect); // clear whole background (w/o widgets)
+
+        var white = "#FFF";
+        var black = "#000";
+        var bord_col = white;
+        var col_off = black;
+
+        var col = new Array(HourCol, MinCol, DayCol, MonCol);  // each #rgb
+        if (g.theme.dark)
         {
-          if (dt_array[line_cnt] >= pot_2[bit_cnt])
-          {
-            dt_array[line_cnt] -= pot_2[bit_cnt];
-            dt_bit_arr[line_cnt][bit_cnt] = 1;
-          }
-          else
-          {
-            dt_bit_arr[line_cnt][bit_cnt] = 0;
-          }
-          bit_cnt--;
+            bord_col = white;
+            col_off = black;
         }
-        
+        else
+        {
+            bord_col = black;   
+            col_off = white;
+        }
+
+        let pwr2 = [1, 2, 4, 8, 16, 32];        // array with powers of 2, because poweroperator '**' doesnt work
```

Comment:
Instead of this you can use bitshifting if you like:

```diff
-pwr2[bit_cnt]
+1 << bit_cnt
```

Reply:
Yes, this is also a good solution. In my opinion the the "LUT"-way may be a bit faster an with only six values not much bigger regarding code size.

Reply:
Sure, we can keep it as-is for now (the bit shift will be faster as there's no memory indirection or bounds checking)